source/dub/packagesuppliers/maven.d

```diff
@@ -92,27 +88,16 @@ class MavenRegistryPackageSupplier : PackageSupplier {
 		auto url = m_mavenUrl~NativePath(packageId~"/maven-metadata.xml");
 
 		logDebug("Downloading maven metadata for %s", packageId);
-		logDebug("Getting from %s", url);
-
 		string xmlData;
-		foreach(i; 0..3) {
-			try {
-				xmlData = cast(string)download(url);
-				break;
-			}
-			catch (HTTPStatusException e)
-			{
-				if (e.status == 404) {
-					logDebug("Maven metadata %s not found at %s (404): %s", packageId, description, e.msg);
-					return Json(null);
-				}
-				else {
-					logDebug("Error getting maven metadata for %s at %s (attempt %s of 3): %s", packageId, description, i + 1, e.msg);
-					if (i == 2)
-						throw e;
-					continue;
-				}
+
+		try
+			xmlData = cast(string)retryDownload(url);
+		catch(HTTPStatusException e) {
+			if (e.status == 404) {
+				logDebug("Maven metadata %s not found at %s (404): %s", packageId, description, e.msg);
+				return Json(null);
 			}
+			else throw e;
 		}
```

Comment:
this now behaves differently. Before it would have just tried to parse an empty xml after 3 fails and put that in the cache, but now it throws after 3 fails. Though this seems like it was a bug before so I think this is a good change here